src/registry.js

```diff
@@ -399,12 +399,15 @@ class AtlasRegistry {
    *   by our registered meters.
    */
   measurements() {
-    let a = [];
+    let a = new Array(this.metersMap.size * 4);
+    let len = 0;
 
     for (let meter of this.metersMap.values()) {
-      Array.prototype.push.apply(a, meter.measure());
+      const measures = meter.measure();
+      len = len + measures.length;
+      Array.prototype.push.apply(a, measures);
```

Comment:
If I understand correctly, `push` creates a new array appending to the end of the one that is passed in, so I don't think this will work correctly. The measures would need to be copied into the correct position for the array `a`.

Reply:
Yeah you are totally right. This is what I meant in my comment while opening the pr, but in my attempt to keep it small changes at a time I overlooked this obvious error.  Thanks! Will update now.